src/StrawberryfieldUtilityService.php

```diff
@@ -28,6 +28,20 @@ class StrawberryfieldUtilityService implements StrawberryfieldUtilityServiceInte
 
   use StringTranslationTrait;
 
+  /**
+   * File system path pattern. Must begin and end with a word character, may have
+   * forward slashes delimiting directory names within the path.
+   */
+  const PATHPATTERN = "/(^(\w)+$)|(^(\w)([\/\w-]*)(\w)$)/";
+
+  /**
+   * S3 file system path pattern.
+   * See \Drupal\strawberryfield\StrawberryfieldUtilityServiceInterface::s3FilePathIsValid
+   * for explanation of the rules.
+   */
+  const S3PATHPATTERN = "/^[a-z0-9][a-z0-9\-]{1,61}[a-z0-9]$/";
```

Comment:
Same here (also super nice)  VALID_S3_PREFIX_PATTERN? (since in S3 paths are normally known as prefixes)

Reply:
Better now I hope.